services/director-v2/src/simcore_service_director_v2/api/routes/computations.py

```diff
@@ -182,60 +182,60 @@ async def get_computation(
     computation_tasks: CompTasksRepository = Depends(
         get_repository(CompTasksRepository)
     ),
-) -> ComputationTaskOut:
-    log.debug("User %s getting computation status for project %s", user_id, project_id)
-    try:
-        # check that project actually exists
-        await project_repo.get_project(project_id)
+) -> ComputationTaskGet:
+    log.debug(
+        "User %s getting computation status for project %s",
+        f"{user_id=}",
+        f"{project_id=}",
+    )
 
-        # NOTE: Here it is assumed the project exists in comp_tasks/comp_pipeline
-        # get the project pipeline
-        pipeline_at_db: CompPipelineAtDB = await computation_pipelines.get_pipeline(
-            project_id
-        )
-        pipeline_dag: nx.DiGraph = pipeline_at_db.get_graph()
+    # check that project actually exists
+    await project_repo.get_project(project_id)
 
-        # get the project task states
-        all_tasks: List[CompTaskAtDB] = await computation_tasks.get_all_tasks(
-            project_id
-        )
-        # create the complete DAG graph
-        complete_dag = create_complete_dag_from_tasks(all_tasks)
+    # NOTE: Here it is assumed the project exists in comp_tasks/comp_pipeline
+    # get the project pipeline
+    pipeline_at_db: CompPipelineAtDB = await computation_pipelines.get_pipeline(
+        project_id
+    )
+    pipeline_dag: nx.DiGraph = pipeline_at_db.get_graph()
 
-        # filter the tasks by the effective pipeline
-        filtered_tasks = [
-            t for t in all_tasks if str(t.node_id) in list(pipeline_dag.nodes())
-        ]
-        pipeline_state = get_pipeline_state_from_task_states(filtered_tasks)
+    # get the project task states
+    all_tasks: List[CompTaskAtDB] = await computation_tasks.get_all_tasks(project_id)
 
-        log.debug(
-            "Computational task status by user %s for project %s is %s",
-            user_id,
-            project_id,
-            pipeline_state,
-        )
+    # filter the tasks by the effective pipeline
+    filtered_tasks = [
+        t for t in all_tasks if str(t.node_id) in list(pipeline_dag.nodes())
```

Comment:
Can you use a set here, or do you have problems with hashable elements?
```suggestion
        t for t in all_tasks if str(t.node_id) in set(pipeline_dag.nodes())
```